src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,160 @@
+            try {
+                switch (commands[0]) {
+                case "bye":
+                    System.out.println("Bye. Hope to see you again soon!\n");
+                    isActive = false;
+                    break;
+                case "list":
+                    for (int i = 0; i < taskCount; i++) {
+                        System.out.println((i + 1) + "." + tasks[i].toString());
+                    }
+                    break;
+                case "mark":
+                    tasks[Integer.parseInt(commands[1]) - 1].markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(tasks[Integer.parseInt(commands[1]) - 1].printTask());
+                    break;
+                case "unmark":
+                    tasks[Integer.parseInt(commands[1]) - 1].markAsUndone();
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println(tasks[Integer.parseInt(commands[1]) - 1].printTask());
+                    break;
+                case "todo":
+                    Todo todo = new Todo(commands[1]);
+                    tasks[taskCount] = todo;
+                    taskCount ++;
+                    System.out.println("Got it I have added this task:");
+                    System.out.println("  " + todo.toString());
+                    System.out.println("Now you have " + taskCount + " tasks in the list.");
+                    break;
+                case "deadline":
+                    addDeadline(commands[1], taskCount, tasks);
+                    break;
+                case "event":
+                    addEvent(commands[1], taskCount, tasks);
+                    break;
+                default:
+                    break;
+                }
+            } catch (IncorrectDeadlineException e) {
+                System.out.println(e.printError());
+            } catch (IncorrectEventException e) {
+                System.out.println(e.printError());
+            }
+        }
+        System.out.println(SYMBOL);
+        input.close();
+    }
+    public static void addDeadline(String command, int taskCount, Task[] tasks) throws IncorrectDeadlineException {
+        if (command.indexOf("/by") == -1) {
+            throw new IncorrectDeadlineException();
+        }
+        String[] message = command.split(" /by", 2);
+        Deadline deadline = new Deadline(message[0], message[1]);
+        tasks[taskCount] = deadline;
+        taskCount++;
+        System.out.println("Got it I have added this task:");
+        System.out.println("  " + deadline.toString());
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+    }
+    public static void addEvent(String command, int taskCount, Task[] tasks) throws IncorrectEventException {
+        if (command.indexOf("/from") == -1 || command.indexOf("/to") == -1) {
```

Comment:
This statement is complex from the code readability point of view. Please refer to the link below to simplify this statement and improve code readability: https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions